airflow/providers/google/cloud/hooks/pubsub.py

```diff
@@ -537,12 +537,10 @@ def acknowledge(
             individual attempt.
         :param metadata: (Optional) Additional metadata that is provided to the method.
         """
-        if ack_ids is not None and messages is None:
-            pass
-        elif ack_ids is None and messages is not None:
-            ack_ids = [message.ack_id for message in messages]
-        else:
+        if not (ack_ids is None) ^ (messages is None):
             raise ValueError("One and only one of 'ack_ids' and 'messages' arguments have to be provided")
+        elif ack_ids is None:
+            ack_ids = [message.ack_id for message in messages]  # type: ignore[union-attr]
```

Comment:
I kind of like the old code actually, it’s a little easier to reason with, but that may be subjective.

Reply:
I revisit to current code and also thought you right and XOR operation actually confuse `mypy`.
I would revert this changes and add comment to empty block.

Reply:
Agree with @uranusjr on this one

Reply:
Agree. The first one is more readable. I think it's not necesary True that "shorter == more readable" . I for example prefer to write classic nested for loop instead of nested for comprehension because I find it much better expressing the intention even if the latter is one-liner.

But in this case I think we have nice util that might make it even more readable - we have `airflow.utils.helpers.exactly_one` (added long time enough to pass >=2.3) and we could use that one making it most readable (even if not that efficient but in this case micro-optimisaatattion might be premature.

Reply:
Return to initial variant. Just add short comments and tests.

Unfortunetly `airflow.utils.helpers.exactly_one` cannot use in this case because first argument (as well as second) for check would be Iterable

Reply:
Damn

Reply:
This helper would be useful in another places

![image](https://user-images.githubusercontent.com/3998685/203174360-14bad73b-fd34-4232-9fdc-71d27bef74fd.png)